File: FPG.Core/Algorithms/Cropping/EvenCroppingAlgorithm.cs
using System;
using System.Text;

using FPG.Interfaces;

namespace FPG.Algorithms
{
    public class EvenCroppingAlgorithm : ICroppingAlgorithm
    {
        public String Crop(String input, Int32 length)
        {
            StringBuilder sb = new StringBuilder();

            for (Int32 i = 0, j = length; i < length; i++, j++)
            {
                if ((j + i) == input.Length)
                {
                    j = 0;
                }

                sb.Append(i % 2 == 0 ? Char.ToUpperInvariant(input[j + i]) : input[j + i]);
            }

            return sb.ToString();
        }
    }
}

File: lib/core/algorithms/cropping/EvenCroppingAlgorithm.dart
import 'package:fpg_mobile/core/interfaces/ICroppingAlgorithm.dart';

class EvenCroppingAlgorithm implements ICroppingAlgorithm {
  @override
  String crop(String input, int length) {
    StringBuffer sb = StringBuffer();

    for (int i = 0, j = length; i < length; i++, j++) {
      if ((j + i) == input.length) {
        j = 0;
      }

      sb.write(i % 2 == 0 ? input[j + i].toUpperCase() : input[j + i]);
    }

    return sb.toString();
  }
}
